src/HttpClient/HttpClientFactory.php

```diff
@@ -123,7 +123,11 @@ private function resolveClient(Options $options)
     {
         if (class_exists(SymfonyHttplugClient::class)) {
             $symfonyConfig = [
-                'max_duration' => self::DEFAULT_HTTP_TIMEOUT,
+                /*
+                 * {@see https://symfony.com/doc/current/http_client.html#dealing-with-network-timeouts}
+                 */
+                'max_duration' => $options->getHttpConnectTimeout() ?? HttpClientFactory::DEFAULT_HTTP_CONNECT_TIMEOUT
```

Comment:
PHPStan says that `getHttpConnectTimeout` is not nullable, but it is. Put it in the baseline with `phpstan --generate-baseline`, it's an issue on all the getters, we should fix it somehow, maybe with array shapes.

Also, if you want to avoid repeating the coalesce every time, I think it's fine to have a temporary variable to reuse on each case.

Reply:
Thanks, all done!